scripts/filemanager.js

```diff
@@ -2765,115 +2837,119 @@ $(function() {
 				}
 
 				if(errorMessage) {
-					$.prompt(lg.upload_failed + "<br>" + errorMessage);
+					$.prompt(fm.lg.upload_failed + "<br>" + errorMessage);
 				} else {
 					// success upload
 					var currentPath = getCurrentPath();
 					getFolderInfo(currentPath);
 					reloadFileTreeNode(currentPath);
-					if(config.options.showConfirmation) {
-						$.prompt(lg.upload_successful_file);
+					if(fm.config.options.showConfirmation) {
+						$.prompt(fm.lg.upload_successful_file);
 					}
 				}
 			})
 
 			.on('fileuploadfail', function(e, data) {
 				// server error 500, etc.
-				$.prompt(lg.upload_failed);
+				$.prompt(fm.lg.upload_failed);
 			});
 	}
 
 	// Loading CustomScrollbar if enabled
 	// Important, the script should be called after calling createFileTree() to prevent bug
-	if(config.customScrollbar.enabled) {
+	if(fm.config.customScrollbar.enabled) {
 		loadCSS('/scripts/custom-scrollbar-plugin/jquery.mCustomScrollbar.min.css');
 		loadJS('/scripts/custom-scrollbar-plugin/jquery.mCustomScrollbar.concat.min.js');
 
-		var csTheme = config.customScrollbar.theme != undefined ? config.customScrollbar.theme : 'inset-2-dark';
-		var csButton = config.customScrollbar.button != undefined ? config.customScrollbar.button : true;
-
-		$(window).load(function() {
-			$filetree.append('<div style="height:3000px"></div>'); // because if #filetree has height equal to 0, mCustomScrollbar is not applied
-			$filetree.mCustomScrollbar({
-				theme: csTheme,
-				scrollButtons: {
-					enable:csButton
-				},
-				advanced: {
-					autoExpandHorizontalScroll: true,
-					updateOnContentResize: true
-				},
-				callbacks: {
-					onInit: function() {
-						createFileTree();
-					}
-				},
-				axis: "yx"
-			});
-
-			$fileinfo.mCustomScrollbar({
-				theme: csTheme,
-				scrollButtons: {
-					enable:csButton
-				},
-				advanced: {
-					autoExpandHorizontalScroll:true,
-					updateOnContentResize: true
-				},
-				axis: "y",
-				alwaysShowScrollbar: 1
-			});
-		});
+		var csTheme = fm.config.customScrollbar.theme != undefined ? fm.config.customScrollbar.theme : 'inset-2-dark';
+		var csButton = fm.config.customScrollbar.button != undefined ? fm.config.customScrollbar.button : true;
+    
+    var fileTreeloaded = jQuery.Deferred();
+    $("#filetree").append('<div style="height:3000px"></div>'); // because if #filetree has height equal to 0, mCustomScrollbar is not applied
+    $("#filetree").mCustomScrollbar({
+      theme:csTheme,
+      scrollButtons:{enable:csButton},
+      advanced:{ 
+        autoExpandHorizontalScroll:true,
+        updateOnContentResize: true 
+      },
+      callbacks:{
+        onInit:function(){ 
+          createFileTree(); 
+          fileTreeloaded.resolve(true);
+        }			
+      },
+      axis: "yx"
+      });
+
```

Comment:
> I tested upload and replace file(s), which seem to work now, although fileTree refresh seems not .. another TODO

Don't worry much about filetree. I'm going to replace it with another, more flexible, plugin and I have started to implement it already.